### PR TITLE
Otlp http log example

### PR DESCRIPTION
**Pull request by esigo**
otlp http log example

## Changes
Adds otlp http log example
Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: examples/common/logs_foo_library/BUILD [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    srcs = [
+        "foo_library.cc",
+    ],
+    hdrs = [
+        "foo_library.h",
+    ],
+    defines = ["BAZEL_BUILD"],
+    deps = [
+        "//api",
+        "//sdk:headers",
```
Comment (lalitb) [inline examples/common/logs_foo_library/BUILD:14]: why do we need sdk::headers, I see this included in earlier PRs too so probably I am missing something :)

File: examples/common/logs_foo_library/foo_library.cc [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+nostd::shared_ptr<trace::Tracer> get_tracer()
+{
+  auto provider = trace::Provider::GetTracerProvider();
+  return provider->GetTracer("foo_library", OPENTELEMETRY_SDK_VERSION);
+}
+
+nostd::shared_ptr<logs::Logger> get_logger()
+{
+  auto provider = logs::Provider::GetLoggerProvider();
+  return provider->GetLogger("foo_library_logger", nostd::span<nostd::string_view>());
```
Comment (lalitb) [inline examples/common/logs_foo_library/foo_library.cc:25]: `return provider->GetLogger("foo_library_logger");` will be much simpler ?

File: examples/common/logs_foo_library/foo_library.cc [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+void foo_library()
+{
+  auto span        = get_tracer()->StartSpan("span 1");
+  auto scoped_span = trace::Scope(get_tracer()->StartSpan("foo_library"));
+  auto ctx         = span->GetContext();
+  auto logger      = get_logger();
+  logger->Log(opentelemetry::logs::Severity::kDebug, "name", "body",
+              std::map<std::string, std::string>(), std::map<std::string, std::string>(),
+              ctx.trace_id(), ctx.span_id(), ctx.trace_flags(),
+              opentelemetry::common::SystemTimestamp());
```
Comment (lalitb) [inline examples/common/logs_foo_library/foo_library.cc:38]: Haven't tested it, but we can use simplify this by using the initializer-list for resources and attributes ( refer API tests for logger for examples ):

``logger->Log(opentelemetry::logs::Severity::kDebug, "name", "body",
             {}, {},
              ctx.trace_id(), ctx.span_id(), ctx.trace_flags(),
              opentelemetry::common::SystemTimestamp());``

File: examples/common/logs_foo_library/BUILD (hunk 1 above)
Comment (esigo) [reply examples/common/logs_foo_library/BUILD:14]: It's needed for `"opentelemetry/sdk/version/version.h"`.

File: examples/otlp/http_log_main.cc [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+#  endif
+
+namespace trace    = opentelemetry::trace;
+namespace nostd    = opentelemetry::nostd;
+namespace otlp     = opentelemetry::exporter::otlp;
+namespace sdklogs  = opentelemetry::sdk::logs;
+namespace logs_api = opentelemetry::logs;
+
+namespace sdktrace = opentelemetry::sdk::trace;
+
```
[the remark this answers is not in the document]
Comment (esigo) [reply examples/otlp/http_log_main.cc:28]: done.

File: examples/common/logs_foo_library/foo_library.cc (hunk 2 above)
Comment (esigo) [reply examples/common/logs_foo_library/foo_library.cc]: done

File: examples/common/logs_foo_library/foo_library.cc (hunk 3 above)
Comment (esigo) [reply examples/common/logs_foo_library/foo_library.cc]: done.

File: examples/common/logs_foo_library/BUILD (hunk 1 above)
Comment (lalitb) [reply examples/common/logs_foo_library/BUILD:14]: ok, thanks for letting me know. Not related to this PR, but we may have to revisit this afterward as ideally instrumentation libraries should have no dependency on SDK.

**Review by lalitb — APPROVED** [review]

**Review by ThomsonTan — APPROVED** [review]

**Pull request merged**